File: RETOS/Reto24/Reto24.py

'''
Crea un programa que simule una carrera de camellos en el que el usuario selecciona uno de los siete camellos
y los camellos avanzan de acuerdo con tiradas aleatorias. El primero que llegue a la meta (casilla 10) ganará 
la carrera.
'''
import random

def carrera_de_camellos():

    camellos = [0] * 7
    meta = 10
    carrera_terminada = False
    
    print("¡Bienvenido a la carrera de camellos!")
    camello_elegido = int(input("Elige tu camello (1 a 7): ")) - 1 
    
    
    while not carrera_terminada:
    
        for i in range(7):
            carrera_visual = "-" * camellos[i] + str(i + 1) + "" * (meta - camellos[i])
            print(carrera_visual, end=" ")
        print()
        

        for i in range(7):
            if random.randint(0, 6) == i:  
                camellos[i] += 1
        

        for i in range(7):
            if camellos[i] >= meta:
                print(f"¡El camello {i + 1} ha llegado a la meta!")
                if i == camello_elegido:
                    print("¡Felicidades! Has ganado.")
                else:
                    print("¡Lo siento! Tu camello no ganó.")
                carrera_terminada = True
                break

carrera_de_camellos()

File: RETOS/Reto24/Reto24.js
    /*
    * Crea un programa que simule una carrera de camellos en el que el usuario selecciona uno de los siete camellos
    * y los camellos avanzan de acuerdo con tiradas aleatorias. El primero que llegue a la meta (casilla 10) ganará
    * la carrera.
    */
const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

function carreraDeCamellos() {
    
    let camellos = Array(7).fill(0);
    const meta = 10;
    let carreraTerminada = false;

    console.log("¡Bienvenido a la carrera de camellos!");
    rl.question("Elige tu camello (1 a 7): ", (camelloElegido) => {
        camelloElegido = parseInt(camelloElegido) - 1;  

        
        while (!carreraTerminada) {
            
            for (let i = 0; i < 7; i++) {
                let carreraVisual = "-".repeat(camellos[i]) + (i + 1) + "".repeat(meta - camellos[i]);
                process.stdout.write(carreraVisual + " ");
            }
            console.log();

            
            for (let i = 0; i < 7; i++) {
                if (Math.floor(Math.random() * 7) === i) {  
                    camellos[i]++;
                }
            }

            
            for (let i = 0; i < 7; i++) {
                if (camellos[i] >= meta) {
                    console.log(`¡El camello ${i + 1} ha llegado a la meta!`);
                    if (i === camelloElegido) {
                        console.log("¡Felicidades! Has ganado.");
                    } else {
                        console.log("¡Lo siento! Tu camello no ganó.");
                    }
                    carreraTerminada = true;
                    break;
                }
            }
        }

        rl.close();
    });
}

carreraDeCamellos();
